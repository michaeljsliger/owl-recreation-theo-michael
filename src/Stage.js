import React, { Component } from 'react';
import STORE from './STORE';
// import STORE from './STORE';
import StageParticipant from './StageParticipant'



class Stage extends Component {
    render() {
        const stageParticipantArray = STORE.filter(el => el.onStage === true).map(el => {
           return (<li key={el.id}><StageParticipant  avatar={el.avatar} name={el.name}/></li>)
        })
        return (
            <ul className={this.props.classNamee}>{stageParticipantArray}</ul>
        )
    }
}

export default Stage;