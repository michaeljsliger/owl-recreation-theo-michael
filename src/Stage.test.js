import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import Stage from './Stage';
import renderer from 'react-test-renderer';

// test('renders learn react link', () => {
//   const { getByText } = render(<App />);
//   const linkElement = getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

describe('Stage testing', () => {
  // smoke test

  it('should render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Stage />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  // snapshot test
  it('should render correctly', () => {
    const tree = renderer
          .create(<Stage />)
          .toJSON();
    expect(tree).toMatchSnapshot();
  })
})