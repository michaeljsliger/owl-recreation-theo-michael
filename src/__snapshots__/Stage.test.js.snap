// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stage testing should render correctly 1`] = `
<ul
  className="stage-list"
>
  <li>
    <div
      className="stage-piece"
    >
      <div
        className="stage-contact"
      >
        <div
          className="stage-contact-header"
        >
          Koren Templeton
        </div>
        <img
          alt="huh"
          className="stage-contact-image"
          src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
        />
      </div>
    </div>
  </li>
  <li>
    <div
      className="stage-piece"
    >
      <div
        className="stage-contact"
      >
        <div
          className="stage-contact-header"
        >
          Axe Kubicka
        </div>
        <img
          alt="huh"
          className="stage-contact-image"
          src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
        />
      </div>
    </div>
  </li>
  <li>
    <div
      className="stage-piece"
    >
      <div
        className="stage-contact"
      >
        <div
          className="stage-contact-header"
        >
          Frank Runciman
        </div>
        <img
          alt="huh"
          className="stage-contact-image"
          src="https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1"
        />
      </div>
    </div>
  </li>
</ul>
`;
